docs/posts/overview-of-ballerina-0.982.0.md

```diff
@@ -0,0 +1,323 @@
+---
+title: Overview of Ballerina 0.982.0
+author: Maryam Ziyad Mohamed
+date: 28 September 2018
+status: Published
+abstract: Introduction to New Features and Changes introduced with Ballerina 0.982.0
+socialmediaimage: //todo
+---
+
+# Improvements and New Features
+The Ballerina 0.982.0 release included new features along with improvements and bug fixes.
+
+## Language & Runtime
+
+### Introduction of the `channel` Type
+
+A new constrained type `channel`, that can be defined only as a top level node, has been introduced for communication 
+between parallel processes in Ballerina programs. Channels can be used for message correlation by sending and receiving 
+messages via different resources to the same channel and can also be used for inter-worker communication and worker 
+synchronization.
+
+Defining a channel constrained by the `JSON` type
+```ballerinalerina
+channel<json> jsonChannel; 
+```
+
+Sending a message to the channel
+    
+```ballerina
+# One of the receivers waiting on the key `key` receives the message.
+# If there is no receiver, the message is stored and execution continues.
+# A receiver can arrive later and fetch the message.
+message -> jsonChannel, key; 
+```
+
+Receive a message from the channel with the given key
+   
+```ballerina
+# Execution waits here if the message is not available.
+json result <- jsonChannel, key;
+```
+
+### Introduction of `abstract` Objects
+
+An abstract object is identified by the `abstract` keyword. It describes the type of each field and each method, but 
+not the implementation of the methods. An abstract object type must not have an object constructor method and does not 
+have an implicit initial value. An abstract object type cannot  be initialized using the object initializer.
+
+```ballerina
+public type Foo abstract object {
+    public string name;
+    public int id;
+
+    function getName() returns string;
+
+    function getID() returns int;
+};
+```
+
+### Introduction of Record Iteration Support
+
+Records are now an iterable type. With this change, the `foreach` statement and iterable operations can now be used 
+with records. When iterating a record, one can iterate either over the fields (i.e., field name and value pair) or only 
+over the field values.
```

Comment:
Minor changes:
"Records are now an iterable type. Therefore, the `foreach` statement and iterable operations can now be used 
with records. When iterating a record, you can either iterate over the fields (i.e., field name and value pair) or iterate only 
over the field values."